theurgy.lic

```diff
@@ -1,326 +1,591 @@
+    fput 'kneel'
+    bput('meditate my prayer chain', 'You clutch')
+    waitfor 'suddenly detaches from your prayer bead chain'
+    fix_standing
+    replace_prayer_chain
+    waitrt?
+    true
+  end
+  
+  def get_prayer_chain
+    case bput('get prayer chain', 'You get', 'I could not find',
+              'What were you referring to', 'already in your inventory')
     when 'I could not find', 'What were you referring to'
-      echo 'No Prayer Chain Found - Skipping Prayer Bead Meditation'
-      return
+      return false
     when 'already in your inventory'
       fput 'remove my prayer chain'
-      worn_chain = true
-    end
-    beads = bput('look at my chain', /^Strung on to the prayer bead chain you see .*\./, 'There are currently no beads on it.')
-    bead_count = list_to_nouns(beads.match(/^Strung on to the prayer bead chain you see (.*)\./).to_a[1])
-    if bead_count.count >= 1
-      fput 'kneel'
-      fput 'meditate my prayer chain'
-      waitfor 'suddenly detaches from your prayer bead chain'
-      fix_standing
-      if worn_chain
-        bput('wear prayer chain', 'You attach', 'You are already wearing that', 'Wear what?')
-      else
-        put_in_container('prayer chain')
-      end
+      @worn_chain = true
     end
-    waitrt?
+    true
   end
 
-  def bathe
-    move 'go pool'
-    get_from_container('sage')
-    fput 'rub sage'
-    pause 1
-    get_from_container('lavender')
-    fput 'rub lavender'
-    waitfor 'A Dark Grove, A Deep Pool'
-    fix_standing
-    move 'out'
+  def count_prayer_beads
+    beads = bput('look at my chain',
+                 /^Strung on to the prayer bead chain you see .*\./,
+                 'There are currently no beads on it.')
+    list_to_nouns(
+      beads.match(/^Strung on to the prayer bead chain you see (.*)\./).to_a[1]).count
   end
-
-  def tithe
-    return unless @settings.tithe
-    delta = Time.now - (UserVars.tithe_timer || Time.now - 14_400)
-    return if delta < 60 * 10
-    return unless withdraw_exact_amount?('5 silver', @hometown)
-
-    UserVars.tithe_timer = Time.now
-    walk_to 741
-    bput('put 5 silver kronar in almsbox', 'You drop', 'But you do not')
-  end
-
-  def pray
-    waitrt?
-    if @favor_god
-      bput("pray #{@favor_god}", @favor_god, 'Lady of healing', 'reward of hard effort', 'sign with your hand', 'god of the Void', 'bane of accursed', 'blessing be upon your love', 'honorable and true', 'glory shine on us', 'fire across the lands', 'madness and pain', 'floods strike down', 'Lady of supreme beauty')
+  
+  def replace_prayer_chain
+    if @worn_chain
+      bput('wear prayer chain', 'You attach',
+           'You are already wearing that', 'Wear what?')
     else
-      bput('pray meraud', 'meraud')
-    end
+      put_in_container('prayer chain')
+    end    
+  end
+
+  def sirese_seed
+    return false unless have_holy_water?
+    walk_to @data['gather_sirese']['id']
+    
+    exit unless DRRoom.npcs.empty? # Invasion check    
```

Comment:
You _might_ want to subtract out `ignored_npcs`. I'm not quite sure if that's right, but it's been a common pattern when checking the room NPCs.

Reply:
Done.